sig-azure/0018-20180711-azure-availability-zones.md

```diff
@@ -0,0 +1,238 @@
+---
+kep-number: 18
+title: Azure Availability Zones
+authors:
+  - "@feiskyer"
+owning-sig: sig-azure
+participating-sigs:
+  - sig-azure
+  - sig-storage
+reviewers:
+  - name: "@khenidak"
+  - name: "@colemickens"
+approvers:
+  - name: "@brendanburns"
+editor: TBD
+creation-date: 2018-07-11
+last-updated: 2018-07-11
+status: provisional
+---
+
+# Azure Availability Zones
+
+## Table of Contents
+
+- [Azure Availability Zones](#azure-availability-zones)
+  - [Summary](#summary)
+  - [Scopes and Non-scopes](#scopes-and-non-scopes)
+    - [Scopes](#scopes)
+    - [Non-scopes](#non-scopes)
+  - [AZ label format](#az-label-format)
+  - [Cloud provider options](#cloud-provider-options)
+  - [Node registration](#node-registration)
+    - [Get by instance metadata](#get-by-instance-metadata)
+    - [Get by Go SDK](#get-by-go-sdk)
+  - [LoadBalancer and PublicIP](#loadbalancer-and-publicip)
+  - [AzureDisk](#azuredisk)
+    - [PVLabeler](#pvlabeler)
+    - [PersistentVolumeLabel](#persistentvolumelabel)
+    - [StorageClass](#storageclass)
+  - [Appendix](#appendix)
+
+## Summary
+
+This proposal aims to add [Azure Availability Zones (AZ)](https://azure.microsoft.com/en-us/global-infrastructure/availability-zones/) support to Kubernetes.
+
+## Scopes and Non-scopes
+
+### Scopes
+
+The proposal includes required changes to support availability zones for various functions in Azure cloud provider and AzureDisk volumes:
+
+- Detect availability zones automatically when registering new nodes and node's label `failure-domain.beta.kubernetes.io/zone` will be replaced with AZ instead of fault domain
+- LoadBalancer and PublicIP will be provisioned with zone redundant
+- `GetLabelsForVolume` interface will be implemented for Azure managed disks and it will also be added to `PersistentVolumeLabel` admission controller so as to support DynamicProvisioningScheduling
+
+### Non-scopes
+
+Provisioning Kubernetes masters and nodes with availability zone support is not included in this proposal. It should be done in the provisioning tools (e.g. acs-engine). Azure cloud provider will auto-detect the node's availability zone if `availabilityZones` option is configured for the Azure cloud provider.
+
+## AZ label format
+
+Currently, Azure nodes are registered with label `failure-domain.beta.kubernetes.io/zone=faultDomain`.
+
+The format of fault domain is numbers (e.g. `1` or `2`), which is in same format with AZ (e.g. `1` or `3`). If AZ is using same format with faultDomain, then there'll be scheduler issues for clusters with both AZ and non-AZ nodes. So AZ will use a different format in kubernetes: `<region>-<AZ>`, e.g. `centralus-1`.
```

Comment:
Maybe you should lead in with some details about what AZ means and what faultDomain means?  Why are they not interchangeable terms (they are both pretty commonly used to mean the same thing)

Reply:
Good catch. This should be definitely addressed.